Update documentation (#48)
idahogurl/vs-code-prettier-eslint: README.md

- `@typescript-eslint/parser@^5.0.1` and `typescript@^4.4.4` **(Only for TypeScript projects)**
- `vue-eslint-parser@^8.0.0` **(Only for Vue projects)**

## Installation

[Download the extension](https://marketplace.visualstudio.com/items?itemName=rvest.vs-code-prettier-eslint) if you haven't already.

**Notes**:

- The [Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) is **not** required.

- The [ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) is **not** required. However, it is needed to have lint errors show while editing your file.

> ## Install dependencies

Reviewer: ```suggestion
## Install Dependencies
```

First we need to install `prettier` and `eslint` as dev dependencies in your project. Run one of the following commands based on your project requirements. The commands listed below use `yarn` but you can also use `npm`. Just replace `yarn add` with `npm i`

```bash 
## minimum requirements
yarn add -D prettier@^2.5.1 eslint@^8.7.0

> ## if your project uses typescript
> yarn add -D prettier@^2.5.1 eslint@^8.7.0 @typescript-eslint/parser@^5.0.1 typescript@^4.4.4
> ```

Reviewer: ```suggestion
### TypeScript Projects
```bash
yarn add -D prettier@^2.5.1 eslint@^8.7.0 @typescript-eslint/parser@^5.0.1 typescript@^4.4.4
```

### Vue Projects
```bash
yarn add -D prettier@^2.5.1 eslint@^8.7.0 vue-eslint-parser@^8.0.0
```
```

Running your files through the [Prettier ESLint CLI](https://github.com/prettier/prettier-eslint-cli) first is a good way to determine if it's the extension, the `prettier-eslint` package, or your configuration. 
## Install dependencies

You can run `eslint --print-config .eslintrc.js` to check your ESLint configuration. This prints out the configuration being used after it applies plugins & extended configurations.
First we need to install `prettier` and `eslint` as dev dependencies in your project. Run one of the following commands based on your project requirements. The commands listed below use `yarn` but you can also use `npm`. Just replace `yarn add` with `npm i`

**Note**: There are known performance issues with quad-core or slower processors. The slowness comes from ESLint CLI which the 'prettier-eslint' package because uses.
```bash 
## minimum requirements
yarn add -D prettier@^2.5.1 eslint@^8.7.0

## Installation
## if your project uses typescript
yarn add -D prettier@^2.5.1 eslint@^8.7.0 @typescript-eslint/parser@^5.0.1 typescript@^4.4.4
```

**Notes**:
## Project settings 

> - The [Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) is not required.

Reviewer: This is actually important. You wouldn't think so but I had people thinking they had to have it installed.

> 1. Open the command palette in vscode by typing:

Reviewer: ```suggestion
1. Open the command palette in VS Code by typing:
```

This is how Microsoft abbreviates Visual Studio Code. See screenshot below.
<img width="1435" alt="Screen Shot 2022-07-20 at 10 08 53 AM" src="https://user-images.githubusercontent.com/10620169/180030856-4fcb70de-6d25-480c-8360-bb6df51efcc5.png">

> 4. **Restart VsCode**  

Reviewer: ```suggestion
4. **Restart VS Code**  
```

<img width="500" alt="troubleshooting" src="https://user-images.githubusercontent.com/10620169/119163604-c9a5cd80-ba18-11eb-91af-fe4d2efe8567.png">

3. Open a JavaScript (`.js` or `.jsx`) or TypeScript (`.ts` or .`tsx`) file
4. Press `CTRL + SHIFT + P` (Windows/Linux) or `CMD + SHIFT + P` (macOS) to open the command palette
5. Start typing `Format Document With` and select `Prettier ESLint`
6. Click on `Output` to open the panel
7. If you see _Error: Cannot find module_, quit and restart Visual Studio Code
8. If restarting did not work:
   - If you saw _Error: Cannot find module '@typescript-eslint/parser'_, view the [TypeScript project](https://github.com/idahogurl/vs-code-prettier-eslint/tree/master/examples/typescript) for a working example.
   - Make sure you have the required packages installed locally (global installations don't work sometimes)

**Note**: Most issues are caused by using an unsupported ESLint version or an invalid ESLint configuration.

Running your files through the [Prettier ESLint CLI](https://github.com/prettier/prettier-eslint-cli) first is a good way to determine if it's the extension, the `prettier-eslint` package, or your configuration. 

> You can run `eslint --print-config .eslintrc.js` to check your ESLint configuration. This prints out the configuration being used after it applies plugins & extended configurations.

Reviewer: ```suggestion
You can run `npx eslint --print-config .eslintrc.js` to check your ESLint configuration. This prints out the configuration being used after it applies plugins & extended configurations.
```

Might as well make changes I've wanted to make for a while 🙂

> **Note**: There are known performance issues with quad-core or slower processors. The slowness comes from ESLint CLI which the 'prettier-eslint' package uses.

Reviewer: ```suggestion
**Note**: There are known performance issues with quad-core or slower processors. The slowness comes from the 'prettier-eslint' package the extension uses.
```

Might as well make changes I've wanted to make for a while 🙂
